### PR TITLE
Remove the condition argument from warning()

### PR DESCRIPTION
**Pull request by walaura**
Step 2 of #16753 

- I've deleted the babel plugin (Woo!) and forked `warning-and-invariant-args` into `warning-args`and `invariant-args`.
- `toWarnDev` and `toLowPriorityWarnDev` didn't actually need any changes as they track the underlying console calls
- Some of the conditionals look wonky, not sure how to improve them without humans
- I don't have java on this laptop yet so can't build but it's installing rn

### CODE REVIEW

File: packages/create-subscription/src/createSubscription.js [hunk 1]
```diff
@@ -37,14 +37,14 @@ export function createSubscription<Property, Value>(
...
   if (__DEV__) {
-    warningWithoutStack(
-      typeof getCurrentValue === 'function',
-      'Subscription must specify a getCurrentValue function',
-    );
-    warningWithoutStack(
-      typeof subscribe === 'function',
-      'Subscription must specify a subscribe function',
-    );
+    if (!(typeof getCurrentValue === 'function')) {
```
Comment (gaearon) [inline packages/create-subscription/src/createSubscription.js]: Let's make a special case and convert these to `!==` negation directly.

File: packages/legacy-events/ResponderTouchHistoryStore.js [hunk 2]
```diff
@@ -95,13 +95,14 @@ function resetTouchRecord(touchRecord: TouchRecord, touch: Touch): void {
...
   invariant(identifier != null, 'Touch object is missing identifier.');
   if (__DEV__) {
-    warningWithoutStack(
-      identifier <= MAX_TOUCH_BANK,
-      'Touch identifier %s is greater than maximum supported %s which causes ' +
-        'performance issues backfilling array locations for all of the indices.',
-      identifier,
-      MAX_TOUCH_BANK,
-    );
+    if (!(identifier <= MAX_TOUCH_BANK)) {
```
Comment (gaearon) [inline packages/legacy-events/ResponderTouchHistoryStore.js]: Same for these. `<=` needs to flip to `>`, `<` needs to flip to `>=`, `>` needs to be `<=`, and `>=` needs to become `<`.

File: packages/react-cache/src/ReactCache.js [hunk 3]
```diff
@@ -64,18 +64,23 @@ function readContext(Context, observedBits) {
 
 function identityHashFn(input) {
   if (__DEV__) {
-    warningWithoutStack(
-      typeof input === 'string' ||
+    if (
+      !(
```
Comment (gaearon) [inline packages/react-cache/src/ReactCache.js]: For more complex conditions like this it's fine to leave them as is.

**Comment by walaura** [comment]
> * Let's detect `if (!(a === b))` early and instead output `if (a !== b)` and possibly the same for `<` and such.
> * Need to fix up `warningWithoutStack.www.js`. It currently re-exports `warning` from WWW. But it has the old API. So `warningWithoutStack.www.js` should now export a function that calls the WWW `warning` with `false` as first argument.

this should be good now but lemme recheck tomorrow

File: packages/shared/forks/lowPriorityWarningWithoutStack.www.js [hunk 4]
```diff
@@ -6,4 +6,9 @@
  */
 
 // This "lowPriorityWarning" is an external module
-export default require('lowPriorityWarning');
+const lowPriorityWarning = require('lowPriorityWarning');
+
+export default function(...args) {
+  // eslint-disable-next-line react-internal/warning-args, react-internal/no-production-logging
+  return lowPriorityWarning(false, args);
```
Comment (gaearon) [inline packages/shared/forks/lowPriorityWarningWithoutStack.www.js]: This one is interesting. Did we change this API in this PR? If not, what's the www API? Has this been wrong the whole time?

**Comment by gaearon** [comment]
I still see the old `!(a === b)`, did you re-run the codemod after those changes?

File: packages/shared/forks/lowPriorityWarningWithoutStack.www.js (hunk 4 above)
Comment (gaearon) [reply packages/shared/forks/lowPriorityWarningWithoutStack.www.js]: actually ignore me, we did change its API

**Comment by MuYunyun** [comment]
👍, the previous usage of warning is `warning(false, 'xxx')`, it increases the difficult of understanding the code.

In my option, it's naturally more to warn in the condition of true.

```js
if (true) {
  warn('xxx')
}
```

and why not write `waring(true, `xxx`)`? :smile:

**Pull request merged**